dumpling: refined as per cn doc (#12064)
pingcap/docs: dumpling-overview.md

</CustomContent>

For detailed usage of Dumpling, use the `--help` option or refer to [Option list of Dumpling](#option-list-of-dumpling).

When using Dumpling, you need to execute the export command on a running cluster.

<CustomContent platform="tidb">

TiDB also provides other tools that you can choose to use as needed.

- For backups of SST files (key-value pairs) or backups of incremental data that are not sensitive to latency, refer to [BR](/br/backup-and-restore-overview.md). 
- For real-time backups of incremental data, refer to [TiCDC](/ticdc/ticdc-overview.md).
- All exported data can be imported back to TiDB using [TiDB Lightning](/tidb-lightning/tidb-lightning-overview.md).

</CustomContent>

> **Note:**
>
> PingCAP previously maintained a fork of the [mydumper project](https://github.com/maxbube/mydumper) with enhancements specific to TiDB. This fork has since been replaced by [Dumpling](/dumpling-overview.md), which has been rewritten in Go, and supports more optimizations that are specific to TiDB. It is strongly recommended that you use Dumpling instead of mydumper.
>

> > For the overview of Mydumper, refer to [v4.0 Mydumper documentation](https://docs.pingcap.com/tidb/v4.0/backup-and-restore-using-mydumper-lightning).

Reviewer: ```suggestion
> For more information on Mydumper, refer to [v4.0 Mydumper documentation](https://docs.pingcap.com/tidb/v4.0/backup-and-restore-using-mydumper-lightning).
```

> 

Reviewer: ```suggestion

Compared to Mydumper, Dumpling has the following improvements:

```

- Support exporting data in multiple formats, including SQL and CSV.
- Support the [table-filter](https://github.com/pingcap/tidb-tools/blob/master/pkg/table-filter/README.md) feature, which makes it easier to filter data.
- Support exporting data to Amazon S3 cloud storage.
- More optimizations are made for TiDB:
    - Support configuring the memory limit of a single TiDB SQL statement.
    - Support automatic adjustment of TiDB GC time for TiDB v4.0.0 and later versions.
    - Use TiDB's hidden column `_tidb_rowid` to optimize the performance of concurrent data export from a single table.
    - For TiDB, you can set the value of [`tidb_snapshot`](/read-historical-data.md#how-tidb-reads-data-from-history-versions) to specify the time point of the data backup. This ensures the consistency of the backup, instead of using `FLUSH TABLES WITH READ LOCK` to ensure the consistency.

## Export data from TiDB or MySQL

### Required privileges

> 

Reviewer: Please add [this paragraph](https://github.com/pingcap/docs-cn/pull/7143/files#diff-d167a2ebb86d802c7a7279da3de5de87e8b616ed0a27755ef0027a8db570c147R61) to line 73:

```
本文假设在 `127.0.0.1:4000` 有一个 TiDB 实例，并且这个 TiDB 实例中有无密码的 root 用户。
```

> 

Reviewer: Two notes are missing from line 152:

```
> - `--sql` 选项仅仅可用于导出 CSV 的场景。
> - 该命令将在要导出的所有表上执行 `select * from <table-name> where id < 100` 语句。如果部分表没有指定的字段，那么导出会失败。
```

Author: 在 L142 行里有的

+ Reduce the value of `--tidb-mem-quota-query` to `8589934592` (8 GB) or lower. `--tidb-mem-quota-query` controls the memory usage of a single query statement in TiDB.
+ Adjust the `--params "tidb_distsql_scan_concurrency=5"` parameter. [`tidb_distsql_scan_concurrency`](/system-variables.md#tidb_distsql_scan_concurrency) is a session variable which controls the concurrency of the scan operations in TiDB.

### Set TiDB GC when exporting a large volume of data (more than 1 TB)

When exporting data from TiDB (more than 1 TB), if the TiDB version is later than or equal to v4.0.0 and Dumpling can access the PD address of the TiDB cluster, Dumpling automatically extends the GC time without affecting the original cluster.

> 

Reviewer: Line 400~410 were removed from the [Chinese version](https://github.com/pingcap/docs-cn/pull/7143/files#diff-d167a2ebb86d802c7a7279da3de5de87e8b616ed0a27755ef0027a8db570c147L332). Please also delete them from English.

Author: OK done
